src/service/back-in-stock.service.ts

```diff
@@ -47,11 +54,54 @@ export class BackInStockService {
         private productVariantService: ProductVariantService,
         private translatorService: TranslatorService,
         private eventBus: EventBus,
-    ) {}
+        @Inject(PLUGIN_INIT_OPTIONS) private options: BackInStockOptions
```

Comment:
Prefer `@Inject(PLUGIN_INIT_OPTIONS) private options: BackInStockOptions` over `BackInStockPlugin.options`. The latter usually works, but you could end up with weird bootstrap scenarios, where `BackInStockPlugin.options` is undefined. With `@Inject` we make NestJS responsible for resolving, which is safer.